src/status_im/ui/screens/wallet_connect/session_proposal/views.cljs

```diff
@@ -0,0 +1,205 @@
+  (letsubs [visible-accounts [:visible-accounts-without-watch-only]
+            dapps-account [:dapps-account]
+            sessions [:wallet-connect/sessions]
+            managed-session [:wallet-connect/session-managed]]
+    (let [{:keys [peer state] :as session} (first (filter #(= (:topic %) topic) sessions))
+          {:keys [accounts]} state
+          {:keys [metadata]} peer
+          {:keys [name icons]} metadata
+          icon-uri (when (and icons (> (count icons) 0)) (first icons))
+          address (last (string/split (first accounts) #":"))
+          account (first (filter #(= (:address %) address) visible-accounts))
+          selected-account (reagent/atom account)]
+      [react/view (styles/proposal-sheet-container)
+       [react/view (styles/proposal-sheet-header)
+        [quo/text {:weight :bold
+                   :size   :large}
+         (i18n/label :t/connection-request)]]
+       [react/image {:style (styles/dapp-logo)
+                     :source {:uri icon-uri}}]
+       [react/view styles/sheet-body-container
+        [react/view styles/proposal-title-container
+         [quo/text {:weight :bold
+                    :size   :large}
+          name]
+         [quo/text {:weight :regular
+                    :size   :large}
+          (i18n/label :t/connected)]]]
+       [account-picker
+        (vector dapps-account)
+        selected-account
+        {:on-press #(do
+                      (re-frame/dispatch [:wallet-connect/manage-app session])
+                      (reset! show-account-selector? true))}]
+       [quo/text {:weight :regular
+                  :color :secondary
+                  :style  styles/message-title}
+        (i18n/label :t/manage-connections)]
+       [react/view (styles/success-button-container)
+        [quo/button
+         {:theme     :accent
+          :on-press  #(do
+                        (reset! show-account-selector? false)
+                        (re-frame/dispatch [:hide-wallet-connect-success-sheet]))}
+         (i18n/label :t/close)]]
+       (when managed-session
+         (if platform/ios?
+           [react/blur-view {:style (styles/blur-view)
+                             :blurAmount 2
+                             :blurType (if (colors/dark?) :dark :light)}]
+           [react/view (styles/blur-view)]))])))
+
+(defview app-management-sheet-view [{:keys [topic]}]
+  (letsubs [sessions [:wallet-connect/sessions]
+            visible-accounts [:visible-accounts-without-watch-only]]
+    (let [{:keys [peer state]} (first (filter #(= (:topic %) topic) sessions))
+          {:keys [accounts]} state
+          {:keys [metadata]} peer
+          {:keys [name icons url]} metadata
+          icon-uri (when (and icons (> (count icons) 0)) (first icons))
+          account-address (last (string/split (first accounts) #":"))
+          selected-account (reagent/atom (first (filter #(= (:address %) account-address) visible-accounts)))]
```

Comment:
atom must be defined  outside the render function, don't remember if its play well with letsubs , should work I guess

Reply:
I would leave this as is if possible, we need to initialize the atom with the session address value and pass it to the flat list to work properly. Any suggestions on how to do this without defining in `let`?

Reply:
we can leave it because its not critical here, but in proper way you have to create extra function pass params in it and in that function define atom outside render function